### ISSUE TITLE
Several properties in the Python API returning lists of int rather than bytearray

### ISSUE THREAD

**Issue by jhhcs** [lief-project/LIEF#833]
**Describe the bug**
There are properties and methods in the Python API that return lists of integers when the more natural type would be `bytearray`, or `bytes`. As far as I can tell from a brief scan of the documentation, this affects most, if not all, properties that return raw binary data from the source file.

**To Reproduce**
For example, use `lief` to parse a resource entry as follows:
```python
def get_icon_group_data(data):
    pe = lief.PE.parse(BytesIO(data))
    for manifest_entry in pe.resources.childs:
        if manifest_entry.id != RSRC.ICON_GROUP.value:
            continue
        return manifest_entry.childs[0].childs[0].content
```

**Expected behavior**
I would expect the function `get_icon_group_data` to return an object that supports the [buffer protocol](https://docs.python.org/3/c-api/buffer.html), i.e. `bytes` or `bytearray`.

**Environment (please complete the following information):**
 - System and Version : Windows 10
 - Target format: PE (but likely others affected)
 - LIEF commit version: 0.12.3-39115d10

**Comment by romainthomas** [comment]
Yes that's because of the Pybind11 default's casting which translates `std::vector<uint8_t>` into a list of integers. I'll see to force returning a `byte` object.
